hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -140,20 +140,6 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
     }
   }
 
-  private Stream<HoodieInstant> getCleanInstantsToArchive() {
-    HoodieTimeline cleanAndRollbackTimeline = table.getActiveTimeline()
-        .getTimelineOfActions(CollectionUtils.createSet(HoodieTimeline.CLEAN_ACTION, HoodieTimeline.ROLLBACK_ACTION)).filterCompletedInstants();
-    return cleanAndRollbackTimeline.getInstants()
-        .collect(Collectors.groupingBy(HoodieInstant::getAction)).values().stream()
-        .map(hoodieInstants -> {
-          if (hoodieInstants.size() > this.maxInstantsToKeep) {
-            return hoodieInstants.subList(0, hoodieInstants.size() - this.minInstantsToKeep);
-          } else {
-            return new ArrayList<HoodieInstant>();
-          }
-        }).flatMap(Collection::stream);
```

Comment:
@nsivabalan @vinothchandar this checks clean and rollback instants separately against min max thresholds. Is the expected logic to check sum of them?

Reply:
if existing logic checks total rollback instants > config value, then wondering why @yihua was facing the issue w/ kafka connect? Do we know if the rollback was completed or was inflight ?

Reply:
@xushiyan : existing logic looks ok to me in general

Reply:
> 

yes that's a separate issue. Here trying to confirm the business logic we intend to set for archiving. In the other archiving logic for commits/deltacommits, it checks the sum.